experimenter/experimenter.py

```diff
@@ -83,75 +84,69 @@ def __init__(self, datasets_dir: str, volumes_dir: str, input_problem_directory=
-        # step_0.add_output('produce')
-        # pipeline_description.add_step(step_0)
-        # step_counter += 1
-
-        # Step 1: dataset_to_dataframe
-        step_1 = PrimitiveStep(
-            primitive=d3m_index.get_primitive('d3m.primitives.data_transformation.dataset_to_dataframe.Common'))
-        step_1.add_argument(name='inputs', argument_type=ArgumentType.CONTAINER, data_reference='inputs.0')
-        step_1.add_output('produce')
-        pipeline_description.add_step(step_1)
-        step_counter += 1
-
-        # Step 2: column_parser
-        step_2 = PrimitiveStep(
-            primitive=d3m_index.get_primitive('d3m.primitives.data_transformation.column_parser.DataFrameCommon'))
-        step_2.add_argument(name='inputs', argument_type=ArgumentType.CONTAINER,
-                            data_reference='steps.{}.produce'.format(step_counter - 1))
-        step_2.add_output('produce')
-        pipeline_description.add_step(step_2)
-        step_counter += 1
-
-        # Step 3: Imputer
-        sk_imputer: PrimitiveBase = d3m_index.get_primitive("d3m.primitives.data_preprocessing.random_sampling_imputer.BYU")
-        step_3 = PrimitiveStep(primitive_description=sk_imputer.metadata.query())
-        step_3.add_argument(name='inputs', argument_type=ArgumentType.CONTAINER,
-                            data_reference = 'steps.{}.produce'.format(step_counter - 1))
-        step_3.add_output('produce')
-        pipeline_description.add_step(step_3)
-        step_counter += 1
-
-        # Step 4: extract_columns_by_semantic_types(targets)
-        step_4 = PrimitiveStep(primitive=d3m_index.get_primitive(
-            'd3m.primitives.data_transformation.extract_columns_by_semantic_types.DataFrameCommon'))
-        step_4.add_argument(name='inputs', argument_type=ArgumentType.CONTAINER, data_reference='steps.0.produce')
-        step_4.add_output('produce')
-        step_4.add_hyperparameter(name='semantic_types', argument_type=ArgumentType.VALUE,
+        # dataset_to_dataframe step
+        step_counter = add_pipeline_step(
+            pipeline_description,
+            step_counter,
+            'd3m.primitives.data_transformation.dataset_to_dataframe.Common',
+            input_data_reference='inputs.0'
+        )
+
+        # column_parser step
+        step_counter = add_pipeline_step(
+            pipeline_description,
+            step_counter,
+            'd3m.primitives.data_transformation.column_parser.DataFrameCommon'
+        )
+
+        # imputer step
+        attributes_step_i = step_counter
+        step_counter = add_pipeline_step(
+            pipeline_description,
+            step_counter,
+            'd3m.primitives.data_preprocessing.random_sampling_imputer.BYU'
+        )
+
+        # extract_columns_by_semantic_types(targets) step
+        extract_targets_step = create_pipeline_step(
```

Comment:
This `create_pipeline_step` helper can be used when additional arguments and hyperparameters need to be added to a step before its added to the pipeline.